### PR TITLE
`get_block_attestns()` & `get_pool_attestns()`

### PR DESCRIPTION
**Pull request by jmcook1186**
implements `get_attestations_from_beacon_block()` and `get_attestations_from_pool()`

### CODE REVIEW

File: src/api_client.rs [hunk 1]
```diff
@@ -231,16 +231,38 @@ impl Client {
     }
 
     pub async fn get_attestations_from_beacon_block(
+        self,
```
Comment (ralexstokes) [inline src/api_client.rs]:
```suggestion
        &self,
```

File: src/types.rs [hunk 2]
```diff
@@ -59,6 +59,18 @@ pub enum BlockId {
     Slot(Slot),
     Root(Root),
 }
+impl fmt::Display for BlockId {
```
Comment (ralexstokes) [inline src/types.rs]: if we merge in #18 and then rebase here, this should go away

File: src/api_client.rs [hunk 3]
```diff
@@ -231,16 +231,38 @@ impl Client {
...
     ) -> Result<Vec<Attestation>, Error> {
-        unimplemented!("")
+        let result: Value<Vec<Attestation>> = self
+            .get(&format!("eth/v1/beacon/blocks/{id}/attestations"))
+            .await?;
+        Ok(result.data)
     }
 
     pub async fn get_attestations_from_pool(
+        self,
```
Comment (ralexstokes) [inline src/api_client.rs]:
```suggestion
        &self,
```

File: src/api_client.rs [hunk 4]
```diff
@@ -231,16 +231,38 @@ impl Client {
...
         slot: Option<Slot>,
         committee_index: Option<CommitteeIndex>,
     ) -> Result<Vec<Attestation>, Error> {
-        unimplemented!("")
+        let path: String = "eth/v1/beacon/pool/attestations".to_string();
+        let target = self.endpoint.join(&path)?;
```
Comment (ralexstokes) [inline src/api_client.rs]: `join` takes `&str` so you can just:
```suggestion
        let path = "eth/v1/beacon/pool/attestations";
        let target = self.endpoint.join(path)?;
```

this is less code and skips the allocation to a `String`

File: src/api_client.rs [hunk 5]
```diff
@@ -231,16 +231,38 @@ impl Client {
...
+        let mut request = self.http.get(target);
+
+        if slot.and(committee_index).is_some() {
+            request = request.query(&[("slot", slot.unwrap())]);
+            request = request.query(&[("committee_index", committee_index.unwrap())]);
+        } else if slot.is_some() && committee_index.is_none() {
+            request = request.query(&[("slot", slot.unwrap())]);
+        } else if slot.is_none() && committee_index.is_some() {
+            request = request.query(&[("committee_index", committee_index.unwrap())]);
+        }
```
Comment (ralexstokes) [inline src/api_client.rs]: the `query` method will expand the `request` additively so we should be able to just handle each param in isolation

I would also suggest pattern matching here, over checking for the boolean `is_*` variant as this is how rustaceans do it :)

```suggestion
        if let Some(slot) = slot { request = request.query(&[("slot", slot)]); }
        if let Some(committee_index) = committee_index { ... }
```

**Review by ralexstokes — COMMENTED** [review]
a few things to clean up but almost there!

**Comment by jmcook1186** [comment]
implemented changes from review, rebased onto main and squashed new commits.

**Review by ralexstokes — APPROVED** [review]

**Pull request merged**